autoload/lightline/bufferline.vim

```diff
@@ -20,8 +21,11 @@ function! s:get_buffer_name(i, buffer)
   elseif s:shorten_path
     let l:name = pathshorten(l:name)
   endif
+  if getbufvar(a:buffer, '&filetype') != 'help' && (getbufvar(a:buffer, '&readonly') || !getbufvar(a:buffer, '&modifiable'))
```

Comment:
This is a large if expression. It would be better to extract it into a function:
```vim
function! s:is_read_only(buffer)
  let l:modifiable = getbufvar(a:buffer, '&modifiable')
  let l:readonly = getbufvar(a:buffer, '&readonly')
  return (l:readonly || !l:modifiable) && getbufvar(a:buffer, '&filetype') != 'help'
endfunction
```